python/ray/tune/checkpoint_manager.py

```diff
@@ -64,28 +64,37 @@ def __init__(self, keep_checkpoints_num, checkpoint_score_attr, delete_fn):
             self._checkpoint_score_attr = checkpoint_score_attr[4:]
         else:
             self._checkpoint_score_attr = checkpoint_score_attr
+
         self.delete = delete_fn
-        self.newest_checkpoint = Checkpoint(Checkpoint.MEMORY, None)
+        self.newest_checkpoint = Checkpoint(Checkpoint.PERSISTENT, None)
+        self.newest_memory_checkpoint = Checkpoint(Checkpoint.MEMORY, None)
         self._best_checkpoints = []
         self._membership = set()
 
     def on_checkpoint(self, checkpoint):
         """Starts tracking checkpoint metadata on checkpoint.
 
-        Sets newest checkpoint. Deletes previous checkpoint as long as it isn't
-        one of the best ones. Also deletes the worst checkpoint if at capacity.
+        Sets the newest checkpoint. For PERSISTENT checkpoints, deletes
+        previous checkpoint if it isn't one of the best ones and the worst
+        checkpoint if at capacity.
```

Comment:
I think the previous wording of best and worst was better.

Reply:
Actually, can you improve this wording? I think this needs to clarify that the newest checkpoint is held and is safe from the eviction policy. Otherwise, checkpoints will be kept according to keep_checkpoint_num and checkpoint_score_attr. 

It would also be important to somehow describe what a theoretical "delete_fn" might do.